Shared/Samples/Monitor changes to layer view state/MonitorChangesToLayerViewStateView.swift

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct MonitorChangesToLayerViewStateView: View {
+    /// A map with a topographic basemap.
+    @State private var map: Map = {
+        let map = Map(basemapStyle: .arcGISTopographic)
+        map.initialViewpoint = Viewpoint(
+            center: Point(x: -13e6, y: 51e5, spatialReference: .webMercator),
+            scale: 2e7
+        )
+        return map
+    }()
+    
+    /// The Satellite (MODIS) Thermal Hotspots and Fire Activity feature layer.
+    @State private var featureLayer: FeatureLayer = {
+        let portalItem = PortalItem(url: .thermalHotspotsAndFireActivity)!
+        let featureLayer = FeatureLayer(item: portalItem)
+        
+        featureLayer.minScale = 1e8
+        featureLayer.maxScale = 6e6
+        
+        return featureLayer
+    }()
+    
+    /// The text representing the view status of the feature layer.
+    @State private var statusText = ""
+    
+    /// A Boolean value indicating whether the feature layer is visible.
+    @State private var layerIsVisible = true
+    
+    var body: some View {
+        MapView(map: map)
+            .onLayerViewStateChanged { layer, layerViewState in
+                // Only checks the view state of the feature layer.
+                guard layer.id == featureLayer.id else { return }
+                
+                // Gets the status from the view state for the layer.
+                statusText = layerViewState.status.label
+            }
+            .overlay(alignment: .top) {
+                Text("Layer view status:\n\(statusText)")
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Toggle(
+                        layerIsVisible ? "Layer Enabled" : "Layer Disabled",
+                        isOn: $layerIsVisible
+                    )
```

Comment:
Did you consider using the switch toggle style? On Catalyst this just appears as an unstyled button.

Reply:
From HIG referring to toggles on mac: 
> Use switches, checkboxes, and radio buttons in the window body, not the window frame. In particular, avoid using these components in a toolbar or status bar.

And for iPhone/iPad:
> Use the switch toggle style only in a list row. 
> Outside of a list, use a button that behaves like a toggle, not a switch.

That is why I instead went with updating the toggle's text based on it's state. I'm not sure that is a great solution either, though, so let me know if you have any better ideas.